rust/lance-encoding/src/encodings/logical/list.rs

```diff
@@ -453,17 +454,20 @@ impl ListFieldEncoder {
                 .map(|page_task| {
                     async move {
                         let page = page_task.await?;
-                        Ok(EncodedPage {
-                            buffers: page.buffers,
-                            column_idx: page.column_idx,
-                            num_rows: page.num_rows,
+                        let array = EncodedArray {
+                            buffers: page.array.buffers,
                             encoding: pb::ArrayEncoding {
                                 array_encoding: Some(pb::array_encoding::ArrayEncoding::List(
                                     Box::new(pb::List {
-                                        offsets: Some(Box::new(page.encoding)),
+                                        offsets: Some(Box::new(page.array.encoding)),
                                     }),
                                 )),
                             },
+                        };
+                        Ok(EncodedPage {
+                            array,
+                            column_idx: page.column_idx,
+                            num_rows: page.num_rows,
                         })
```

Comment:
minor nit:
```suggestion
                        Ok(EncodedPage { array, ..page })
```

Reply:
Added, thanks (surprised clippy doesn't suggest this yet)+